sceptre_cdk_handler/cdk_builder.py

```diff
@@ -0,0 +1,214 @@
+        # that will always be correct...
+        result = self._subprocess_run(
+            command,
+            env=env,
+            shell=True,
+            stdout=sys.stderr,
+            check=True
+        )
+
+        return result
+
+    def _get_envs(self) -> Dict[str, str]:
+        """
+        Obtains the environment variables to pass to the subprocess.
+
+        Sceptre can assume roles, profiles, etc... to connect to AWS for a given stack. This is
+        very useful. However, we need that SAME connection information to carry over to CDK when we
+        invoke it. The most precise way to do this is to use the same session credentials being used
+        by Sceptre for other stack operations. This method obtains those credentials and sets them
+        as environment variables that are passed to the subprocess and will, in turn, be used by
+        SAM CLI.
+
+        The environment variables dict created by this method will inherit all existing
+        environment variables in the current environment, but the AWS connection environment
+        variables will be overridden by the ones for this stack.
+
+        Returns:
+            The dictionary of environment variables.
+        """
+        envs = self._environment_variables.copy()
+        envs.pop("AWS_PROFILE", None)
+        # Set aws environment variables specific to whatever AWS configuration has been set on the
+        # stack's connection manager.
+        credentials: Credentials = self._connection_manager._get_session(
+            self._connection_manager.profile,
+            self._connection_manager.region,
+            self._connection_manager.iam_role
+        ).get_credentials()
+        envs.update(
+            AWS_ACCESS_KEY_ID=credentials.access_key,
+            AWS_SECRET_ACCESS_KEY=credentials.secret_key,
+            # Most AWS SDKs use AWS_DEFAULT_REGION for the region
+            AWS_DEFAULT_REGION=self._connection_manager.region,
+            # CDK frequently uses CDK_DEFAULT_REGION in its docs
+            CDK_DEFAULT_REGION=self._connection_manager.region,
+            # cdk-assets requires AWS_REGION to determine what region's STS endpoint to use
+            AWS_REGION=self._connection_manager.region
+        )
+
+        # There might not be a session token, so if there isn't one, make sure it doesn't exist in
+        # the envs being passed to the subprocess
+        if credentials.token is None:
+            envs.pop('AWS_SESSION_TOKEN', None)
+        else:
+            envs['AWS_SESSION_TOKEN'] = credentials.token
+
+        return envs
+
+    def _only_asset_is_template(self, asset_artifacts: aws_cdk.cx_api.AssetManifestArtifact):
+        manifest_contents = asset_artifacts.contents
+        if manifest_contents.docker_images:
+            return False
+
+        keys = list(manifest_contents.files.keys())
+        expected_template = f'{self.STACK_LOGICAL_ID}.template.json'
+        return keys == [expected_template]
```

Comment:
I only see this method used in an if statement (if self._only_asset_is_template(asset_artifacts)) in this file.  why does it need to return anything besides a boolean?

Reply:
This method _does_ return a boolean. I think you might be misreading it. It returns True if the only keys on the files manifest are the expected template stack name, False if otherwise.

Reply:
ahh ok.